helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableNode.java

```diff
@@ -287,10 +281,21 @@ private void assignNewBatch(Collection<AssignableReplica> replicas) {
     }
   }
 
-  private boolean addToAssignmentRecord(AssignableReplica replica,
-      Map<String, Set<String>> currentAssignments) {
-    return currentAssignments.computeIfAbsent(replica.getResourceName(), k -> new HashSet<>())
-        .add(replica.getPartitionName());
+  /**
+   * @return True if the replica has been added to record successfully. False if the replica already
+   * exists on this node.
+   */
+  private boolean addToAssignmentRecord(AssignableReplica replica) {
+    if (null == _currentAssignments.computeIfAbsent(replica.getResourceName(), k -> new HashMap<>())
```

Comment:
Add some comment here - say that you're doing a null comparison and relying on the behavior of putIfAbsent(). 
I personally do not like this style and think it's awkward. It's not necessarily more readable or concise than doing a if (!contains()) check. I'd personally just go with a contains check. Or at least add more comments so it reads better.

Reply:
Sure, I agree. I think I went too far, haha. Will change it.